mxcube3/routes/SampleCentring.py

```diff
@@ -192,156 +188,92 @@ def get_image_data():
+    resp.status_code = 200
+    return resp
 
 
-@mxcube.route("/mxcube/api/v0.1/sampleview/centring/<centring_id>/moveto", methods=['PUT'])
-def move_to_centred_position(centring_id):
+@mxcube.route("/mxcube/api/v0.1/sampleview/shapes", methods=['POST'])
+def update_shapes():
     """
-    Move to the given centred position.
-        :parameter id: centred position identifier, integer
+    Retrieve all the stored centred positions.
+        :response Content-type: application/json, the stored centred positions.
         :statuscode: 200: no error
         :statuscode: 409: error
     """
-    motor_positions = [d['motor_positions'] for d in mxcube.diffractometer.savedCentredPos if d.get('posId') == int(centring_id)]
-    if 'kappa' in motor_positions[0]:
-        motor_positions[0].pop('kappa')
-    if 'kappa_phi' in motor_positions[0]:
-        motor_positions[0].pop('kappa_phi')
-    mxcube.diffractometer.move_to_motors_positions(
-        copy.deepcopy(motor_positions[0]))
-    logging.getLogger('HWR.MX3').info('[Centring] moved to Centring Position')
-    return Response(status=200)
+    resp = Response(status=409)
+    params = request.get_json()
+    shape_data = from_camel(params.get("shapeData", {}))
+
+    # Get the shape if already exists
+    shape = mxcube.shapes.get_shape(params["id"])
 
+    # If shape does not exist add it
+    if not shape:
+        refs, t = shape_data.pop("refs", []), shape_data.pop("t", "")
 
-@mxcube.route("/mxcube/api/v0.1/sampleview/centring", methods=['GET'])
-def get_centring_positions():
+        # Shape does not have any refs, create a new Centered position
+        if not refs:
+            x, y = shape_data["screen_coord"]
+            mpos = mxcube.diffractometer.get_centred_point_from_coord(x, y)
+            shape = mxcube.shapes.add_shape_from_mpos([mpos], (x, y), t)
+        else:
+            shape = mxcube.shapes.add_shape_from_refs(refs, t)
+
+    # shape will be none if creation failed, so we check if shape exists
+    # before setting additional parameters
+    if shape:
+        shape.update_from_dict(shape_data)
+        shape_dict = shape.as_dict()
+        resp = jsonify(to_camel(shape_dict))
+        resp.status_code = 200
+
+    return resp
+
+
+@mxcube.route("/mxcube/api/v0.1/sampleview/shapes/<sid>", methods=['DELETE'])
+def delete_shape(sid):
     """
     Retrieve all the stored centred positions.
         :response Content-type: application/json, the stored centred positions.
         :statuscode: 200: no error
         :statuscode: 409: error
     """
-    aux = {}
-    for pos in mxcube.diffractometer.savedCentredPos:
-        aux.update({pos['posId']: pos})
-        pos_x, pos_y = mxcube.diffractometer.motor_positions_to_screen(
-            pos['motor_positions'])
-        aux[pos['posId']].update({'x': pos_x, 'y': pos_y})
-    resp = jsonify(aux)
-    resp.status_code = 200
-    return resp
+    mxcube.shapes.delete_shape(sid)
+    return Response(status=200)
```

Comment:
Yes, we can act on individual shapes if we like to, but we don't need to in general. We want to remove all shapes when we mount or unmount a sample.